src/ui/CategoryFacet/CategoryFacet.ts

```diff
@@ -0,0 +1,132 @@
+    exportGlobally({
+      CategoryFacet
+    });
+  };
+
+  static ID = 'CategoryFacet';
+
+  static options: CategoryFacetOptions = {
+    field: ComponentOptions.buildFieldOption({ required: true }),
+    title: ComponentOptions.buildLocalizedStringOption({
+      defaultValue: l('NoTitle')
+    }),
+    id: ComponentOptions.buildStringOption({ postProcessing: (value, options: CategoryFacetOptions) => value || (options.field as string) })
+  };
+
+  private categoryValueRoot: CategoryValueRoot;
+  private categoryFacetTemplates: CategoryFacetTemplates;
+  private facetHeader: Dom;
+  private waitElement: Dom;
+  public queryStateAttribute: string;
+
+  public categoryValueRootModule = CategoryValueRoot;
+
+  constructor(public element: HTMLElement, public options: CategoryFacetOptions, bindings?: IComponentBindings) {
+    super(element, 'CategoryFacet', bindings);
+    this.options = ComponentOptions.initComponentOptions(element, CategoryFacet, options);
+
+    this.categoryFacetQueryController = new CategoryFacetQueryController(this);
+    this.categoryFacetTemplates = new CategoryFacetTemplates();
+    this.categoryValueRoot = new this.categoryValueRootModule($$(this.element), this.categoryFacetTemplates, this);
+    this.buildFacetHeader();
+    this.initQueryStateEvents();
+  }
+
+  public getChildren(): CategoryValue[] {
+    return this.categoryValueRoot.getChildren();
+  }
+
+  public disable() {
+    super.disable();
+    $$(this.element).addClass('coveo-hidden');
+  }
+
+  public hide() {
+    $$(this.element).addClass('coveo-hidden');
+  }
+
+  public show() {
+    $$(this.element).removeClass('coveo-hidden');
+  }
+
+  public showWaitingAnimation() {
+    if (this.waitElement.el.style.visibility == 'hidden') {
+      this.waitElement.el.style.visibility = 'visible';
+    }
+  }
+
+  public hideWaitAnimation() {
+    if (this.waitElement.el.style.visibility == 'visible') {
+      this.waitElement.el.style.visibility = 'hidden';
+    }
+  }
+
+  private buildFacetHeader() {
+    this.waitElement = $$('div', { className: 'coveo-category-facet-header-wait-animation' }, SVGIcons.icons.loading);
+    SVGDom.addClassToSVGInContainer(this.waitElement.el, 'coveo-category-facet-header-wait-animation-svg');
+    this.waitElement.el.style.visibility = 'hidden';
+
+    const titleSection = $$('div', { className: 'coveo-category-facet-title' }, this.options.title);
+    this.facetHeader = $$('div', { className: 'coveo-category-facet-header' }, titleSection);
+    $$(this.element).prepend(this.facetHeader.el);
+    this.facetHeader.append(this.waitElement.el);
+  }
+
+  private handleQueryStateChanged(data: IAttributesChangedEventArg) {
+    if (this.listenToQueryStateChange) {
+      let path = data.attributes[this.queryStateAttribute];
+      if (!Utils.isNullOrUndefined(path) && _.isArray(path) && path.length != 0) {
+        this.categoryValueRoot.setActivePath(path);
+      }
+    }
+  }
+
+  public updatePath(path: string[]) {
```

Comment:
Put public method(s) together in the class file.

Also, the API for this component is kinda lacking. 

From an external user perspective I'd expect some methods to select and deselect values, to navigate to a path, to go up a level to immediate parent, go up to "root" values, reset the component, list the currently selected path, list the available values, etc. etc...